pkg/api/pod/testing/make.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/utils/ptr"
+)
+
+// Tweak is a function that modifies a Pod.
+type Tweak func(*api.Pod)
+type TweakContainer func(*api.Container)
+
+// MakePod helps construct Pod objects (which pass API validation) more
+// legibly and tersely than a Go struct definition.  By default this produces
```

Comment:
I was not paying too much attention to the comments since I was planning to add them once it was agreed upon. I also realized that many of the functions are only used in 1 location or another. I was debating whether it should be moved when not really reused.